app/models/manageiq/providers/ansible_tower/shared/automation_manager/tower_api.rb

```diff
@@ -76,9 +83,12 @@ def update_in_provider(params)
     with_provider_object do |provider_object|
       provider_object.update_attributes!(params)
     end
+
     if respond_to?(:refresh_in_provider)
-      refresh_in_provider
+      refresh_success = refresh_in_provider
+      self.class.send('notify_refresh_in_provider', manager_id, params, refresh_success)
```

Comment:
Why not make this public so we don't have to use `send`?

Reply:
Ok, so why are this new method and `notify` both private class methods that we want to call from an instance?  If they were all instance methods, you can call them directly without `send` or `self` and the "internal consumption" purpose can be maintained.  If you want to call them from both class scope and instance scope, they're not private methods.

Reply:
Note, I'm not saying to fix this all right now in this PR but the usage of `send` indicates improper visibility of this method.

Reply:
The method visibility is not all that wrong. It is accessed only from the `TowerApi` class itself or its instance. It must be called via `send` though, because it is defined as private inside the `ClassMethods` module.

This is however not my invention: I used the style already used in the class. The `notify`, `queue` etc. methods are in that space too and so are they called using `send`.